proposals/0000-textinput-onpaste.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Add onPaste to TextInput
+author:
+    - Abdelhafidh Belalia
+date: 2024-07-18
+---
+
+# RFC0000: Add onPaste to TextInput
+
+## Summary
+
+Add `onPaste` prop to `TextInput` component
+
+## Basic example
+
+```js
+<TextInput onPaste={() => console.log("User pressed Paste")} />
+```
+
+## Motivation
+
+Detect `Paste` event on TextInput. In our use case detecting paste event can be used to support image pasting.
+
+## Detailed design
+
+iOS:
+
+1. In `RCTUITextField`/`RCTUITextView` override the `paste:` method to call the delegate adapater `[didPaste]`
+2. Implement the `didPaste` method on the delegate adapater and call the delegate input `[textInputDidPaste]`
+3. Implement the `textInputDidPaste` method on the delegate input to disptach the `onPaste` event
+
+Android:
+
+1. In `ReactEditText` override the `onTextContextMenuItem` method to call the `PasteWatcher` insatnce
+2. In `ReactTextInputManager` create and set the `PasteWatcher` instance for the view (`ReactEditText`)
+3. Implement the `PasteWatcher` class to to disptach the `onPaste` event
+
+There is a PR for that change already https://github.com/facebook/react-native/pull/45425
+
+## Drawbacks
+
+-   onPaste callback timing is not the same in iOS vs Android. In iOS `onPaste` is called first then `onChange`. While in Android it's the other way around.
```

Comment:
Any `onChange`  triggered by paste would ideally come after paste event, per https://w3c.github.io/clipboard-apis/#otherevents

I think `onTextContextMenuItem()` might happen before change on Android?

Reply:
> I think `onTextContextMenuItem()` might happen before change on Android?

Indeed! I just found out that I was dispatching the event after calling `super.onTextContextMenuItem` (i.e. after pasting the text). Dispatching the event first fixes this inconsistency


https://github.com/user-attachments/assets/01f74705-90fe-40fb-8045-c1849aa83ea8